llvm/lib/Target/AMDGPU/AMDGPUAttributor.cpp

```diff
@@ -1323,6 +1298,60 @@ struct AAAMDGPUNoAGPR
 
 const char AAAMDGPUNoAGPR::ID = 0;
 
+/// The final check and update of the attribute 'amdgpu-waves-per-eu' based on
+/// the determined 'amdgpu-flat-work-group-size' attribute. We can't do this
+/// during attributor run because the two attributes grow in opposite direction,
+/// we should not use any intermediate value to calculate waves per eu until we
+/// have a determined flat workgroup size.
```

Comment:
I still believe the wording here is misleading. In our last (long) discussion, I tried to propose a different view on the same issue, as "grows in the opposite direction" is not a conceptual problem. I would recommend looking back at that conversation, maybe you can expand on the above a bit to help people understand the conceptual issue.

Reply:
How do you like the new version?